protocols/kad/src/query.rs

```diff
@@ -137,9 +155,23 @@ impl QueryPool {
     ) where
         T: Into<KeyBytes> + Clone,
         I: IntoIterator<Item = Key<PeerId>>,
+    {
+        self.continue_iter_closest_inner(id, target, peers, info, self.config.replication_factor);
+    }
+
+    fn continue_iter_closest_inner<T, I>(
```

Comment:
Can be consolidated with `continue_iter_closest` by adding `Option<NonZeroUsize>` since the function is only used in this crate.

If no `num_results` is provided, use `self.config.replication_factor`.